client/src/pages/Details.js

```diff
@@ -30,12 +35,15 @@ export default function Details({ member, updateMember, members }) {
     event.preventDefault();
 
     if (isValidEntry(entry)) {
+      //Add valid entry to entries array and set entry to initial state afterwards
       const newEntry = { ...entry, id: uuid4() };
       const memberEntries = [...entries, newEntry];
       setEntries(memberEntries);
       setEntry(initialEntry);
 
-      const updatedMember = { ...member, entries: memberEntries };
+      //Update the newMember with the new entries and send the updatedMember to App.js
+      const updatedMember = { ...newMember, entries: memberEntries };
+      setNewMember(updatedMember);
```

Comment:
Both these functions seem to always be called together. Maybe useful to encapsulate the calling in a separate function. This new function would then make sure both functions are called.